### PR TITLE
Expose onRenderColorCell

### PR DESCRIPTION
**Pull request by bhavinkotak07**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)
SwatchColorPicker doesn't have any onRender function. With this change we can customize the UI of the Color picker to show a color and font picker combo. Below is the screenshot which shows the proposed change.
image

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx [hunk 1]
```diff
@@ -148,6 +148,11 @@ export class SwatchColorPickerBase extends React.Component<ISwatchColorPickerPro
     );
   }
 
+  private _onRenderItem = (item: IColorCellProps, index: number): JSX.Element => {
```
Comment (RupamSarkar) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx:151]: Do we need the `index` argument in this function?

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts [hunk 2]
```diff
@@ -74,7 +74,10 @@ export interface ISwatchColorPickerProps {
    * If `id` and `color` are unspecified, cells are no longer being focused.
    */
   onCellFocused?: (id?: string, color?: string) => void;
-
+  /**
+   * Custom render function for the color cell
+   */
+  onRenderColorCell?: IRenderFunction<IColorCellProps>;
```
Comment (RupamSarkar) [inline packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts:80]: Do we have any UT for SwatchColorPicker ? If yes, then can we add one UT for with this new prop ?

File: packages/office-ui-fabric-react/src/utilities/ButtonGrid/ButtonGrid.types.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ export interface IButtonGridProps extends React.TableHTMLAttributes<HTMLTableEle
   /**
    * Custom renderer for the individual items.
    */
-  onRenderItem: (item: any, index: number) => JSX.Element;
+  onRenderItem: (item: any, index: number) => JSX.Element | null;
```
Comment (shagra-ms) [inline packages/office-ui-fabric-react/src/utilities/ButtonGrid/ButtonGrid.types.ts:26]: Can you please explain why change in return type is required?

Comment (bhavinkotak07) [reply packages/office-ui-fabric-react/src/utilities/ButtonGrid/ButtonGrid.types.ts:26]: The onRenderItem function invokes onRenderColorCell which is of IRenderFunction type and it return JSX.Element | null. 
Not changing the return type will make compiler complain.

**Review by shagra-ms — APPROVED** [review]

File: packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts (hunk 2 above)
Comment (bhavinkotak07) [reply packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.types.ts:80]: Yes we have a UT for SwatchColorPicker and I have added a scenario for it.

**Review by ecraig12345 — COMMENTED** [review]
Can you please make this change in the react-next version of SwatchColorPicker as well?

**Review by dzearing — APPROVED** [review]
While we would like to rework ColorSwatchPicker, this change is reasonable as a stop gap for reconfiguring the swatches.

**Review by dzearing — CHANGES_REQUESTED** [review]
Needs to be updated in react-next as well.

**Comment by shagra-ms** [comment]
Thanks for the inputs @dzearing and @ecraig12345. We will do the needful.

**Pull request closed without merging**